src/Compilers/VisualBasic/Portable/Syntax/VisualBasicSyntaxTree.ParsedSyntaxTree.vb

```diff
@@ -143,13 +159,37 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
 
                 Return New ParsedSyntaxTree(
-                    Me._lazyText,
-                    Me._encodingOpt,
-                    Me._checksumAlgorithm,
+                    _lazyText,
+                    _encodingOpt,
+                    _checksumAlgorithm,
                     path,
-                    Me._options,
-                    Me._root,
-                    Me._isMyTemplate)
+                    _options,
+                    _root,
+                    _isMyTemplate,
+                    cloneRoot:=True,
+                    diagnosticOptions:=_diagnosticOptions)
+            End Function
+
+            Public Overrides Function WithDiagnosticOptions(options As IEnumerable(Of (String, ReportDiagnostic))) As SyntaxTree
+                If options Is Nothing Then
+                    options = Array.Empty(Of (String, ReportDiagnostic))
+                End If
+
+                If _diagnosticOptions.IsEmpty AndAlso options.IsEmpty() Then
+                    Return Me
+                End If
+
+                Return New ParsedSyntaxTree(_lazyText,
+                                            _encodingOpt,
+                                            _checksumAlgorithm,
+                                            _path,
+                                            _options,
+                                            _root,
+                                            _isMyTemplate,
+                                            cloneRoot:=True,
+                                            ImmutableDictionary.CreateRange(
+                                                StringComparer.OrdinalIgnoreCase,
```

Comment:
Previously you used StringOrdinalComparer.Instance.  Having different ways of referencing the same thing was pretty confusing to me.  

Also, it would make sense to put the helper to make the dictionary out of the keyvaluepairs in the ParseOptions type, so we don't have to repeat and remember this in every case.

Reply:
They're not the same thing. This one is case insensitive.

Reply:
Sorry, it's unclear ot me what the actual keys are here.  I don't see this actually used in any code.  As such, i don't actually understand why we need different case sensitivity options...  can you clarify?

Reply:
I added a comment to clarify what the result is. As to why the diagnostic ID may be case insensitive or sensitive depending on the language, see VisualBasicDiagnosticFilter.Filter. It explicitly constructs a new dictionary for compilation SpecificDiagnosticOptions that is case-insensitive on diagnostic options.  I'm just continuing the pattern.

Reply:
Wow!  Well, if that's the case, then that def makes sense.  But comments are definitely welcome here as that would be totally unobvious from reading the code :)